scripts/jenkins/cloud/ansible/roles/ardana_reboot/tasks/reboot_nodes.yml

```diff
@@ -28,13 +28,22 @@
 
 - name: Run post-reboot playbook on deployer
   command: |
-    ansible-playbook _ardana-post-reboot.yml --limit ARD-SVC--first-member  -e '{{ ardana_extra_vars|to_json }}'
+    ansible-playbook _ardana-post-reboot.yml --limit {{ ardana_deployer }}  -e '{{ ardana_extra_vars|to_json }}'
   args:
     chdir: "{{ ardana_scratch_path }}"
 
-- name: Reboot non-deployer nodes serially
+- name: Reboot non-deployer controller nodes serially, if any
   command: |
     ansible-playbook ardana-reboot.yml --limit {{ item }} -e '{{ ardana_extra_vars|to_json }}'
   args:
     chdir: "{{ ardana_scratch_path }}"
-  loop: "{{ ardana_nodes.stdout_lines }}"
+  loop: "{{ ardana_controllers }}"
+  when: (ardana_controllers | count) > 0
+
+- name: Reboot compute nodes serially, if any
+  command: |
+    ansible-playbook ardana-reboot.yml --limit {{ item }} -e '{{ ardana_extra_vars|to_json }}'
+  args:
+    chdir: "{{ ardana_scratch_path }}"
+  loop: "{{ ardana_computes }}"
+  when: (ardana_computes | count) > 0
```

Comment:
same here

Reply:
So technically Ansible should skip iterating over the loop when the list is empty, but I have seen weirdness in the past with Ansible 1.9, 2.0-2.3 and I think it was 2.6 but not 2.5 or 2.7 where in a case like the above if you are registering the result that the "status" for the action in the registered fact may not be what you expected, e.g. if the list is empty, the status might be success even though it is skipped...

Count it as me being paranoid about ensuring the status is reported as skipped if we are truely skipping it ;-)

Reply:
ok!